docs/pages/blog/masonry-post.md

```diff
@@ -0,0 +1,141 @@
+
+`<Masonry />` receives two special props: `columns` and `spacing`. `columns` is the number of columns of the masonry, while `spacing` is the gap between items of the masonry. Both props can be a fixed value, such as `3` and `4`, or responsive values, such as `{ xs: 3, sm: 4, md: 5 }` and `[ xs: 3, sm: 4, md: 5]`. Further details can be found in its [API documents](https://next.material-ui.com/api/masonry/).
+
+The key to this implementation is to set the row height to `1px`. This can be done by setting the height of implicit grid rows to `0px` and the row gap to `1px`. In more detail, the following code, `grid-auto-rows: 0; row-gap: 1`, does the job.
+
+The property `grid-template-columns` should change in response to the `columns` prop. For example, if `columns` is `3`, `<Masonry />` will have the following property: `grid-template-columns: repeat(3, 1fr)`.
+
+The property `column-gap` should change in response to the `spacing` prop. For example, if `spacing` is `2`, `<Masonry />` will have the following property: `column-gap: theme.spacing(2)`. It is worth noting here that `spacing` is a factor of the theme's spacing. You may ask why we only change the `column-gap`. What about the `row-gap`? As we already discussed, `row-gap: 1` should always be held in order for this implementation to work. Otherwise, `<MasonryItem />` cannot span in a multitude of `1px`.
+
+This is pretty much the gem of `<Masonry />`. It is much simpler relative to `<MasonryItem />`, which we are now moving on to look in detail.
+
+### `<MasonryItem />`
+
+This component is a CSS grid item that wraps around one and only child. Through the use of `React.Children.only()`, we enforce that only one child is passed to the component. MUI users can always create more `<MasonryItem />` to add more children.
+
+`<MasonryItem />` receives two special props: `columnSpan` and `defaultHeight`. `columnSpan` is the number of columns taken up by the component, while `defaultHeight`, which is provided for optional server-side rendering, is the initial height of the component. Both props receive a fixed value. Further details can be found in its [API documents](https://next.material-ui.com/api/masonry-item/).
+
+`<MasonryItem />` utilises a Web API called [ResizeObserver API](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver). It detects changes to the dimensions of an element. We use this API to observe the child passed to `<MasonryItem />`. Every time the child's height changes, which is commonly caused by window resizing or zooming, these two properties, `grid-row-end` and `padding-bottom`, also get updated. `grid-row-end` changes so that the height of `<MasonryItem />`, by spanning a correct number of rows, matches its child's height. `padding-bottom` changes so that the vertical gap between `<MasonryItem />`s stays correct.
```

Comment:
```suggestion
`<MasonryItem />` utilizes a Web API called [ResizeObserver API](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver). It detects changes to the dimensions of an element. We use this API to observe the child passed to `<MasonryItem />`. Every time the child's height changes, which is commonly caused by window resizing or zooming, these two properties, `grid-row-end` and `padding-bottom`, also get updated. `grid-row-end` changes so that the height of `<MasonryItem />`, by spanning a correct number of rows, matches its child's height. `padding-bottom` changes so that the vertical gap between `<MasonryItem />`s stays correct.
```